### ISSUE TITLE
"Unable to start your application" After expo upgrade from SDK 39 to SDK 40

### ISSUE THREAD

**Issue by eliottguillaumin** [expo/expo#11683]
### Summary of Issue 
The app crashes on start on both platforms using expo Client after I've upgraded my project to SDK40.
I had made sure to change the imports accordingly but it shows me an error no one had yet reported.

### Environment - output of `expo diagnostics` & the platform(s) you're targeting

```
Expo CLI 4.0.17 environment info:
    System:
      OS: macOS 10.15.7
      Shell: 5.7.1 - /bin/zsh
    Binaries:
      Node: 14.15.3 - /usr/local/bin/node
      npm: 6.14.9 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.10.1 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: iOS 14.3, DriverKit 20.2, macOS 11.1, tvOS 14.3, watchOS 7.2
    IDEs:
      Android Studio: 3.3 AI-182.5107.16.33.5314842
      Xcode: 12.3/12C33 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^40.0.0 => 40.0.0
      react: 16.13.1 => 16.13.1
      react-dom: 16.13.1 => 16.13.1
      react-native: https://github.com/expo/react-native/archive/sdk-40.0.1.tar.gz => 0.63.2
      react-native-web: ~0.13.12 => 0.13.18
    npmGlobalPackages:
      expo-cli: 4.0.17
    Expo Workflow: managed
```

Targetting both iOS and Android
### Reproducible Demo

The shell output when run: 

```
Unable to start your application. Please refer to https://expo.fyi/no-registered-application for more information.
- node_modules/react-native/Libraries/LogBox/LogBox.js:117:10 in registerWarning
- node_modules/react-native/Libraries/LogBox/LogBox.js:63:8 in warnImpl
- node_modules/react-native/Libraries/LogBox/LogBox.js:36:4 in console.warn
- node_modules/expo/build/environment/react-native-logs.fx.js:18:4 in warn
- node_modules/react-native/Libraries/ReactNative/AppRegistry.js:200:6 in runApplication
- node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:416:4 in __callFunction
- node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:109:6 in __guard$argument_0
- node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:364:10 in __guard
- node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:108:4 in callFunctionReturnFlushedQueue
* [native code]:null in callFunctionReturnFlushedQueue

ReferenceError: Can't find variable: ls
- node_modules/react-native/Libraries/LogBox/LogBox.js:148:8 in registerError
- node_modules/react-native/Libraries/LogBox/LogBox.js:59:8 in errorImpl
- node_modules/react-native/Libraries/LogBox/LogBox.js:33:4 in console.error
- node_modules/expo/build/environment/react-native-logs.fx.js:27:4 in error
- node_modules/react-native/Libraries/Core/ExceptionsManager.js:104:6 in reportException
- node_modules/react-native/Libraries/Core/ExceptionsManager.js:171:19 in handleException
- node_modules/react-native/Libraries/Core/setUpErrorHandling.js:24:6 in handleError

ReferenceError: Can't find variable: ls
- node_modules/react-native/Libraries/LogBox/LogBox.js:148:8 in registerError
- node_modules/react-native/Libraries/LogBox/LogBox.js:59:8 in errorImpl
- node_modules/react-native/Libraries/LogBox/LogBox.js:33:4 in console.error
- node_modules/expo/build/environment/react-native-logs.fx.js:27:4 in error
- node_modules/react-native/Libraries/Core/ExceptionsManager.js:104:6 in reportException
- node_modules/react-native/Libraries/Core/ExceptionsManager.js:171:19 in handleException
- node_modules/react-native/Libraries/Core/setUpErrorHandling.js:24:6 in handleError
- node_modules/regenerator-runtime/runtime.js:63:36 in tryCatch
- node_modules/regenerator-runtime/runtime.js:293:29 in invoke
- node_modules/regenerator-runtime/runtime.js:63:36 in tryCatch
- node_modules/regenerator-runtime/runtime.js:154:27 in invoke
- node_modules/regenerator-runtime/runtime.js:164:18 in PromiseImpl.resolve.then$argument_0
- node_modules/react-native/node_modules/promise/setimmediate/core.js:37:13 in tryCallOne
- node_modules/react-native/node_modules/promise/setimmediate/core.js:123:24 in setImmediate$argument_0
- node_modules/react-native/Libraries/Core/Timers/JSTimers.js:130:14 in _callTimer
- node_modules/react-native/Libraries/Core/Timers/JSTimers.js:181:14 in _callImmediatesPass
- node_modules/react-native/Libraries/Core/Timers/JSTimers.js:441:30 in callImmediates
- node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:387:6 in __callImmediates
- node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:135:6 in __guard$argument_0
- node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:364:10 in __guard
- node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:134:4 in flushedQueue
* [native code]:null in flushedQueue
* [native code]:null in invokeCallbackAndReturnFlushedQueue
```

My App.js:

```js
import React, { useState } from "react";
import { StyleSheet } from "react-native";

import { NavigationContainer, useNavigation } from "@react-navigation/native";

import AuthNavigator from "./app/navigation/AuthNavigator";
import navigationTheme from "./app/navigation/navigationTheme";
import OfflineNoticeBar from "./app/components/OfflineNoticeBar";
import AuthContext from "./app/auth/context";
import timemath from "./app/utility/timemath";
import ParseApi from "./app/api/parseAPI";
import AppLoading from 'expo-app-loading';

import {
  useFonts,
  Inter_600SemiBold,
  Inter_500Medium,
} from "@expo-google-fonts/inter";
import {
  Lato_400Regular,
  Lato_400Regular_Italic,
  Lato_700Bold,
} from "@expo-google-fonts/lato";
import { Oswald_700Bold, Oswald_600SemiBold } from "@expo-google-fonts/oswald";
import DashboardTabNavigator from "./app/navigation/DashboardTabNavigator";


export default App = () => {
  const [user, setUser] = useState({});
  const [isReady, setIsReady] = useState(false);
  ParseApi.startParse();

  let [fontsLoaded] = useFonts({
    InterSB: Inter_600SemiBold,
    InterM: Inter_500Medium,
    LatoR: Lato_400Regular,
    LatoRi: Lato_400Regular_Italic,
    LatoB: Lato_700Bold,
    OswaldSB: Oswald_600SemiBold,
    OswaldB: Oswald_700Bold,
  });

  const restoreUser = async () => {
    const user = await ParseApi.getCurrentUser();
    timemath.setTimeDelayAsync();
    if (user) setUser(user);
  };
  
  if (!isReady || !fontsLoaded) {
    return <AppLoading startAsync={restoreUser} onFinish={() => setIsReady(true)} />;
  }

  return (
    <AuthContext.Provider value={{ user, setUser }}>
      <OfflineNoticeBar />
      <NavigationContainer theme={navigationTheme}>
        {user ? <DashboardTabNavigator /> : <AuthNavigator />}
      </NavigationContainer>
    </AuthContext.Provider>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
});

```

My package.json:
```json
{
  "main": "node_modules/expo/AppEntry.js",
  "scripts": {
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "web": "expo start --web",
    "eject": "expo eject"
  },
  "dependencies": {
    "@expo-google-fonts/inter": "^0.1.0",
    "@expo-google-fonts/lato": "^0.1.0",
    "@expo-google-fonts/oswald": "^0.1.0",
    "@react-native-async-storage/async-storage": "^1.13.2",
    "@react-native-community/masked-view": "0.1.10",
    "@react-native-community/netinfo": "5.9.7",
    "@react-navigation/bottom-tabs": "^5.11.2",
    "@react-navigation/native": "^5.8.10",
    "@react-navigation/stack": "^5.12.8",
    "apisauce": "^2.0.0",
    "asyncStorage": "^0.1.0",
    "asyncstorage": "^1.5.0",
    "dayjs": "^1.9.6",
    "expo": "^40.0.0",
    "expo-app-loading": "^1.0.1",
    "expo-av": "~8.7.0",
    "expo-blur": "~8.2.2",
    "expo-constants": "~9.3.3",
    "expo-document-picker": "~8.4.1",
    "expo-font": "~8.4.0",
    "expo-image-picker": "~9.2.0",
    "expo-linear-gradient": "~8.4.0",
    "expo-location": "~10.0.0",
    "expo-media-library": "~10.0.0",
    "expo-network": "~2.4.0",
    "expo-notifications": "~0.8.2",
    "expo-permissions": "~10.0.0",
    "expo-secure-store": "~9.3.0",
    "expo-status-bar": "~1.0.3",
    "expo-video-player": "^1.6.1",
    "formik": "^2.2.5",
    "jwt-decode": "^3.1.2",
    "lottie-react-native": "~2.6.1",
    "moment": "^2.29.1",
    "parse": "^2.19.0",
    "react": "16.13.1",
    "react-dom": "16.13.1",
    "react-native": "https://github.com/expo/react-native/archive/sdk-40.0.1.tar.gz",
    "react-native-draggable-view": "0.0.6",
    "react-native-elements": "^3.0.0-alpha.1",
    "react-native-expo-image-cache": "^4.1.0",
    "react-native-gesture-handler": "~1.8.0",
    "react-native-maps": "0.27.1",
    "react-native-progress": "^4.1.2",
    "react-native-qrcode-svg": "^6.0.6",
    "react-native-reanimated": "~1.13.0",
    "react-native-safe-area-context": "3.1.9",
    "react-native-screens": "~2.15.0",
    "react-native-svg": "12.1.0",
    "react-native-web": "~0.13.12",
    "undefined": "expo/videoplayer",
    "yup": "^0.31.0"
  },
  "devDependencies": {
    "@babel/core": "~7.9.0"
  },
  "private": true
}

```

 
### Steps to Reproduce

Use expo upgrade on a project and use npm start (don't know how I can detail it more, I have seriously no idea why it does that)

### Expected Behavior vs Actual Behavior

Expected: App starts normally like before the upgrade

Actual: App crashes

Please do not close this issue, the reason the steps may be vague are because this big came out of nowhere on a project I've spent months on, if I were to detail all the steps involved in getting this result I'd be leaking protected code

**Comment by brentvatne** [comment]
there is some code in your app that is attempting to use the variable `ls` and it is not defined

> ReferenceError: Can't find variable: ls

**Comment by eliottguillaumin** [comment]
I could not run with remote debugging because the app crashed before I could open the menu in Expo to enable remote debugging.
I thankfully found a stray 
`
ls;
`
in a portion of code deep inside imports by taking hours to comment on all import statements to figure out which was at fault...
The expo upgrade was apparently not at fault, but this type of error paralyzed my work for 2 full days.
If any Expo dev stumbles upon this issue please make it so that it shows what file is at fault when something like this happens

**Comment by IcodeNet** [comment]
I was in the same situation for hours. In my case the message was
 `ReferenceError: Can't find variable: StyleSheet`
 after searching globally in my src files I found that an import was removed from one of my files.
 Once I added ` import { StyleSheet } from 'react-native' ` everything started working again.
 
 I do agree though that giving the name of the offending file is probably a good idea. 
 PS. Thanks for the amazing tooling and framework
